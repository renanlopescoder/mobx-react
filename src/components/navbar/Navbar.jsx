import React from 'react';
import { Link } from 'react-router-dom';
import { AppBar, IconButton, Navigation, FontIcon, IconMenu, MenuItem, MenuDivider, Avatar } from 'react-toolbox';

import style from './style.css';
import color from '../../shared/colors.css';

class Navbar extends React.Component {
  render() {
    return (
      <AppBar theme={style}>
        <IconMenu icon='menu' position='topLeft' menuRipple>
          <Link to='/' className={style.link}>
            <MenuItem value='home' icon='home' caption='Home' />
          </Link>
          <Link to='/projects/new' className={style.link}>
            <MenuItem value='register' icon='create' caption='Register' />
          </Link>
          <Link to='/projects' className={style.link}>
            <MenuItem value='projects' icon='format_align_left' caption='Projects' />
          </Link>
          <Link to='/' className={style.link}>
            <MenuItem value='configuration' icon='settings' caption='Configuration' />
          </Link>
          <MenuDivider />
          <Link to='/logout' className={style.link}>
            <MenuItem value='logout' icon='reply' caption='Logout' />
          </Link>
        </IconMenu>
        <h1>
          MobX React
        </h1>
        <Navigation type="horizontal">
          <Avatar className={[style.userInfo, color.primary]} icon="person" title='Renan Lopes'/>
        </Navigation>
      </AppBar>
    )
  };
};

export default Navbar;
