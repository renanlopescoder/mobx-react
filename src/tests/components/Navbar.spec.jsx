import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Navbar from '../../../src/components/navbar/Navbar'

describe('components/<Navbar />', () => {
  it('should render the app navbar', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    )
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})
