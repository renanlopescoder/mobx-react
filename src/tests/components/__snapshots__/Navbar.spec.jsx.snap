// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/<Navbar /> should render the app navbar 1`] = `
<header
  className=""
  data-react-toolbox="app-bar"
>
  <div
    className={undefined}
  >
    <h1>
      <a
        href="/"
        onClick={[Function]}
      />
    </h1>
    <nav
      className=""
      data-react-toolbox="navigation"
    >
      <a
        className={undefined}
        href="/logout"
        onClick={undefined}
      >
        <button
          className=""
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
          >
            exit_to_app
          </span>
        </button>
      </a>
    </nav>
  </div>
</header>
`;
